I3dm fixes

lasalvavida opened the pull request:
Fixes from https://github.com/AnalyticalGraphicsInc/3d-tiles/pull/121.

gltfByteLength is omitted and computed from the total byteLength.

Includes updates to sample tiles.

lilleyse commented on Source/Scene/Instanced3DModel3DTileContent.js:
```diff
@@ -257,6 +249,8 @@ define([
         //>>includeEnd('debug');
         byteOffset += sizeOfUint32;
 
+        var gltfByteLength = byteLength - 32 - featureTableJSONByteLength - featureTableBinaryByteLength - batchTableJSONByteLength - batchTableBinaryByteLength;
```
If you haven't already it's worth taking a look at how `Batched3DModel3DTileContent` computes this, it's a bit simpler.

lilleyse commented:
Also update `Cesium3DTilesTester.generateInstancedTileBuffer`. This probably explains the failing tests.

lasalvavida commented:
Updated. I think the composite tile content may not have gotten updated with the promise fix, so I fixed that here as well to correct the failing tests I was getting.

lilleyse commented on Source/Scene/Composite3DTileContent.js:
```diff
@@ -215,7 +215,7 @@ define([
                 content.initialize(arrayBuffer, byteOffset);
                 this._contents.push(content);
-                contentPromises.push(content.readyPromise);
+                contentPromises.push(content._readyPromise);
```
I think this is actually fine how it was, since we want the actual promises and not the deferreds. Both versions work because deferreds are also thennable.

lasalvavida replied:
You are correct, will revert.

lilleyse commented:
The rest of the changes look good and all tests pass.

lasalvavida commented:
Updated
